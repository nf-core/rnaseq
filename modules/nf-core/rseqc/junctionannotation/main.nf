process RSEQC_JUNCTIONANNOTATION {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)
    path  bed

    output:
    tuple val(meta), path("*.xls")         , emit: xls
    tuple val(meta), path("*.r")           , emit: rscript
    tuple val(meta), path("*.log")         , emit: log
    tuple val(meta), path("*.junction.bed"), optional:true, emit: bed
    tuple val(meta), path("*.Interact.bed"), optional:true, emit: interact_bed
    tuple val(meta), path("*junction.pdf") , optional:true, emit: pdf
    tuple val(meta), path("*events.pdf")   , optional:true, emit: events_pdf
    path  "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    junction_annotation.py \\
        -i $bam \\
        -r $bed \\
        -o $prefix \\
        $args \\
        2> >(grep -v 'E::idx_find_and_load' | tee ${prefix}.junction_annotation.log >&2)

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(junction_annotation.py --version | sed -e "s/junction_annotation.py //g")
    END_VERSIONS
    """
}
