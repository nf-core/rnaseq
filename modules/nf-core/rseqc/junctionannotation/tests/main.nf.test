nextflow_process {

    name "Test Process RSEQC_JUNCTIONANNOTATION"
    script "../main.nf"
    process "RSEQC_JUNCTIONANNOTATION"
    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/junctionannotation"

    test("sarscov2 - paired end [bam]") {

        when {
            process {
                """
                input[0] = [
                            [ id:'test', single_end: false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.log).match("log") },
                { assert snapshot(process.out.versions).match("versions") },
                { assert process.out.xls.get(0).get(1) ==~ ".*/test.junction.xls" },
                { assert process.out.rscript.get(0).get(1) ==~ ".*/test.junction_plot.r" }
            )
        }

    }

}
