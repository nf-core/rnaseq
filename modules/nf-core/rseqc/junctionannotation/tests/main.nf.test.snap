{
    "log": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.junction_annotation.log:md5,ef37d06d169a1adbeec23fddf82aee2b"
                ]
            ]
        ],
        "timestamp": "2023-12-14T18:56:11.636775765"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,091865c0ebd6de262c9b0968b5771091"
            ]
        ],
        "timestamp": "2023-12-14T17:08:02.340160345"
    }
}